### ISSUE TITLE
Question about the APD (average pose distance) metric calculation

### ISSUE THREAD

**Issue by fusheng-ji** [aigc3d/LAM#11]
Hello,

I have a question about APD metric calculation in the paper. As I read the issue in [OTAvatar](https://github.com/theEricMa/OTAvatar/issues/11#issuecomment-1534002382), they compute the APD with the L1 loss on concatenation of roation and translation.

Here is my implement:

First I extract angles and translation coeffs from Deep3DFaceRecon:
`angles = pred_coeff[224:227]  # euler angles for pose`
`translation = pred_coeff[254:257]  # translation`
`pose_coeff = np.concatenate([angles, translation], 0)`

Then I calculate L1 loss between pose_coeff of predction and ground truth.
`apd_loss = np.mean(np.abs(pred_pose_coeff - gt_pose_coeff))`

However, after I getting the average metrics on HDTF and VFHQ evaluation, the values of APD are much lower than number in paper. 

Could you help me with that?

Many Thanks

**Comment by weihaosky** [comment]
If I remember correctly, APD metric is only for angles.

**Comment by fusheng-ji** [comment]
Hello @weihaosky,

Thanks for your clarification. I will compute it with only angles.

Really appreciate your help.

**Comment by yinian1234567** [comment]
> 你好，
> 
> 谢谢你的澄清。我只用角度计算。
> 
> 非常感谢你的帮助。


Hello, @fusheng-ji I also met some problem in reproducing the metric.Here is my implement:

First,I referred the Inference scripts in [Deep3DFaceRecon_pytorch](https://github.com/sicxu/Deep3DFaceRecon_pytorch/blob/9167c7136aebef0c54d3ac74f4b0396222a491c1/test.py)  as the issue in [OTAvatar](https://github.com/theEricMa/OTAvatar/issues/27#issuecomment-1801217498)

This file used  [model.test()](https://github.com/sicxu/Deep3DFaceRecon_pytorch/blob/9167c7136aebef0c54d3ac74f4b0396222a491c1/test.py#L63)   to get facial coefficient.

And I found that the function test as followed:[test](https://github.com/sicxu/Deep3DFaceRecon_pytorch/blob/9167c7136aebef0c54d3ac74f4b0396222a491c1/models/base_model.py#L155-L163)
And the forward function is as followed:[forward](https://github.com/sicxu/Deep3DFaceRecon_pytorch/blob/9167c7136aebef0c54d3ac74f4b0396222a491c1/models/facerecon_model.py#L134-L142)

So to get facial coefficient,I use 
"""
        model.forward()
        coef = model.pred_coeffs_dict
        exp = coef["exp"]
        pose = coef["angle"]
"""
By the way I got AED and APD values that are much lower than the given results.

I want to know how you managed to reproduce the results from the paper?

Looking forward to your reply
